src/feature/nodelist/microdesc.c

```diff
@@ -535,19 +537,27 @@ microdesc_cache_reload(microdesc_cache_t *cache)
 
   journal_content = read_file_to_str(cache->journal_fname,
                                      RFTS_IGNORE_MISSING, &st);
-  if (journal_content) {
+  if (journal_content && st.st_size) {
     cache->journal_len = (size_t) st.st_size;
     warn_if_nul_found(journal_content, cache->journal_len, 0,
                       "reading microdesc journal");
+
+    /* Replace all NUL bytes in journal_content with space character. */
+    for (size_t i = 0; i < (size_t)st.st_size; i++)
+      if (journal_content[i] == '\0')
+        journal_content[i] = ' ';
```

Comment:
Hm. I'm not sure I think this is the right solution to the problem. Ideally there should never be any NUL bytes around here, so can't we instead pretend that the file is at EOF when we have the first NUL?